From 8545f8bf5f4c4bb373e0805508b7789984672453 Mon Sep 17 00:00:00 2001
From: Alon Levy <alevy@redhat.com>
Date: Wed, 18 Apr 2012 12:24:28 +0300
Subject: [PATCH] qxl-render: fix broken vnc+spice since commit f934493

Notify any listeners such as vnc that the displaysurface has been
changed, otherwise they will segfault when first accessing the freed old
displaysurface data.

Signed-off-by: Alon Levy <alevy@redhat.com>
---
 hw/qxl-render.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hw/qxl-render.c b/hw/qxl-render.c
index 835dc5e..180b8f9 100644
--- a/hw/qxl-render.c
+++ b/hw/qxl-render.c
@@ -127,6 +127,7 @@ static void qxl_render_update_area_unlocked(PCIQXLDevice *qxl)
             qemu_resize_displaysurface(vga->ds,
                     qxl->guest_primary.surface.width,
                     qxl->guest_primary.surface.height);
+            dpy_resize(vga->ds);
         }
     }
     for (i = 0; i < qxl->num_dirty_rects; i++) {
